go-controller/pkg/allocator/udn/allocator.go

```diff
@@ -0,0 +1,51 @@
+package udn
+
+import (
+	"fmt"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type Allocator struct {
+	watchFactory factory.NodeWatchFactory
+	nodeName     string
+}
+
+func New(watchFactory factory.NodeWatchFactory, nodeName string) *Allocator {
+	return &Allocator{
+		watchFactory: watchFactory,
+		nodeName:     nodeName,
+	}
+}
+
+func (a *Allocator) calculateIDsFromNetwork(idName, networkName string, base, limit, cardinality uint) (uint, error) {
+	if cardinality < 1 {
+		return 0, fmt.Errorf("invalid arguments, cardinality '%d' has to be bigger than '1'", cardinality)
+	}
+	if limit <= base {
+		return 0, fmt.Errorf("invalid arguments, limit '%d' has to be bigger than base '%d'", limit, base)
+	}
+	node, err := a.watchFactory.GetNode(a.nodeName)
```

Comment:
its weird to be doing this inside the allocator by passing entire watch factory, I think instead of passing the networkname just pass the networkID into this function?